File: apps/ios/Barns/BarnsApp/Domain/Entities/ConsultationDraft.swift
import Foundation

struct ConsultationDraft: Equatable, Identifiable, Sendable {
    let id: String
    let productItemId: String?
    let topic: String
    let category: ConsultationCategory
    let urgency: ConsultationUrgency
    let body: String
    let status: ConsultationDraftStatus
    let createdAt: Date
    let updatedAt: Date
    let imageUrl: URL?
}

enum ConsultationCategory: String, Equatable, Sendable {
    case maintenance
    case care
    case replacement
    case other
}

enum ConsultationUrgency: String, Equatable, Sendable {
    case low
    case normal
    case high
}

enum ConsultationDraftStatus: String, Equatable, Sendable {
    case draft
    case archived
}

File: apps/android/app/src/main/java/com/barns/app/domain/model/ConsultationDraft.kt
package com.barns.app.domain.model

import java.time.Instant

data class ConsultationDraft(
    val id: String,
    val productItemId: String?,
    val topic: String,
    val category: ConsultationCategory,
    val urgency: ConsultationUrgency,
    val body: String,
    val status: ConsultationDraftStatus,
    val createdAt: Instant,
    val updatedAt: Instant,
    val imageUrl: String?,
)

enum class ConsultationCategory { MAINTENANCE, CARE, REPLACEMENT, OTHER }

enum class ConsultationUrgency { LOW, NORMAL, HIGH }

enum class ConsultationDraftStatus { DRAFT, ARCHIVED }
